tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc

```diff
@@ -3689,3 +3689,96 @@ TEST_F(ConvectionCoefficientsFixture, testTARPNaturalConvectionAlgorithm)
+        thisMaterial_1->AbsorpThermalFront = 0.1;
+        thisMaterial_1->Roughness = Material::SurfaceRoughness::Rough;
+        thisMaterial_1->Name = "Roof_Material";
+        thisMaterial_1->group = Material::Group::Regular;
+
+        // set environment air conditions
+        state->dataEnvrn->OutBaroPress = 101325.0; // Pa
+        state->dataEnvrn->OutHumRat = 0.0075;      // kgH20/kgDryAir
+
+        // set convection coefficient calc parameters
+        int const SurfNum = 1;
+        Real64 const SurfTemp = 20.0;
+        Real64 const AirTemp = 15.0;
+        Real64 const WindAtZ = 2.0; // m/s
+        Real64 const WindDirect = 0.0;
+        Real64 RoofArea = RoofGeo.Area;
+        Real64 RoofPerimeter = RoofGeo.Perimeter;
+
+        // test 1: calc exterior convection coefficient
+        // non zero roof area and perimeter
+        Real64 Hf_result1 = CalcClearRoof(*state, SurfNum, SurfTemp, AirTemp, WindAtZ, WindDirect, RoofArea, RoofPerimeter);
+        EXPECT_NEAR(10.42, Hf_result1, 0.01); //
+        EXPECT_EQ(state->dataSurface->Surface(1).ExtBoundCond, EnergyPlus::DataSurfaces::ExternalEnvironment);
+        EXPECT_TRUE(compare_err_stream(""));
+
+        // test 2: calc exterior convection coefficient
+        // reset roof area and perimeter to 0.0 to generate error message
+        RoofArea = 0.0;
+        RoofPerimeter = 0.0;
+        Real64 Hf_result2 = CalcClearRoof(*state, SurfNum, SurfTemp, AirTemp, WindAtZ, WindDirect, RoofArea, RoofPerimeter);
+        EXPECT_NEAR(9.9999, Hf_result2, 0.0001);
+        EXPECT_EQ(state->dataSurface->Surface(1).ExtBoundCond, EnergyPlus::DataSurfaces::ExternalEnvironment);
+        // expect severe error message
+        EXPECT_TRUE(has_err_output(false));
+        std::string error_string = delimited_string(
+            {"   ** Severe  ** CalcClearRoof: Convection model not evaluated (bad value for distance to roof edge)\n   **   ~~~   ** "
+             "Value for distance to roof edge =0.000\n   **   ~~~   ** Occurs for surface named = Roof Surface\n   **   ~~~   ** "
+             "Convection surface heat transfer coefficient set to 9.999 [W/m2-K] and the simulation continues"});
+        EXPECT_TRUE(compare_err_stream(error_string, true));
+
+        // test 3: calc exterior convection coefficient
+        // Ext boundary conditions set to OtherSideCondModeledExt and expect no error
+        state->dataSurface->Surface(1).ExtBoundCond = EnergyPlus::DataSurfaces::OtherSideCondModeledExt;
+        RoofArea = 0.0;
+        RoofPerimeter = 0.0;
+        Real64 Hf_result3 = CalcClearRoof(*state, SurfNum, SurfTemp, AirTemp, WindAtZ, WindDirect, RoofArea, RoofPerimeter);
+        EXPECT_DOUBLE_EQ(9.9999, Hf_result3);
+        EXPECT_EQ(state->dataSurface->Surface(1).ExtBoundCond, EnergyPlus::DataSurfaces::OtherSideCondModeledExt);
+        EXPECT_TRUE(compare_err_stream(""));
+    }
+}
```

Comment:
Added unit test that demonstrates the error messages expected depending on the boundary condition of an exterior surface.

Reply:
This is a **fantastic** unit test @Nigusse, great work!  Clear and obvious demonstration of the problem with multiple scenarios, not a bunch of IDF getting in the way, good stuff!